plat/xen/Linker.uk

```diff
@@ -1,5 +1,7 @@
 ifeq (x86_64,$(CONFIG_UK_ARCH))
 XEN_LDFLAGS-y += -Wl,-m,elf_x86_64
+else ifeq (arm64,$(CONFIG_UK_ARCH))
+XEN_LINK_LIBGCC_FLAG := -lgcc
```

Comment:
I understand that you essentially adapted `KVM_LINK_LIBGCC_FLAG ` to Xen. However, just a question, is this really named accordingly in the first place (`KVM_LINK_LIBGCC_FLAG` included)? This is a flag that is directly passed to the linker. Shouldn't this have been `XEN_LDFLAGS-y += -lgcc`? (Same for the KVM one).

Reply:
@mogasergiu , the `KVM_LINK_LIBGCC_FLAG` was introduced and used in three commits:

- commit [`eddaa9b24`](https://github.com/unikraft/unikraft/commit/eddaa9b24) by Jia He
- commit [`08a9ebf34`](https://github.com/unikraft/unikraft/commit/08a9ebf34) by @marcrittinghaus 
- commit [`46d98734c`](https://github.com/unikraft/unikraft/commit/46d98734c) by you

I didn't check to see why there is a separate variable defined (`KVM_LINK_LIBGCC_FLAG`), instead of simply adding it to the `KVM_FLAGS` variable.

If it would be better to add it to the `KVM_FLAGS` variable (seems OK to me)., I would approve and merge this PR and then create a separate PR that fixes both KVM and Xen.

Reply:
Yes, you are right. The last two commits are just copy-paste though. It just randomly jumped in my face now, as a reviewer, that this name looks misaligned with the others.

> If it would be better to add it to the KVM_FLAGS variable (seems OK to me)., I would approve and merge this PR and then create a separate PR that fixes both KVM and Xen.

Sure, makes sense to me, if it makes sense to the other as well. That being said, I do not have any strong opinions regarding this, I was just merely pointing it out :D.

Reply:
@michpappas , as you're the assignee of this PR, what do you think? Is it OK if we kept it like this and I opened an issue and a pull request that moves the `-lgcc` flag from `KVM_LINK_LIBGCC_FLAG` to `KVM_FLAGS`?

Reply:
I agree with @mogasergiu here, and especially on that it's good to have `-lgcc` appended to `LDFLAGS-y` in a separate line, as one can easily trace the rationale through the commit history. @razvand the change is tiny so it would be really quick to update this PR with an additional commit for KVM, but if you think otherwise we can also do as you suggest. So if you'd still rather do it later, please let me know and I'll proceed to trigger the merge.

Reply:
@michpappas , I updated the pull request with a new commit to have this fixed. Please check it out.